--- pesign-113/src/pesign.sysvinit.in.orig	2019-05-10 20:53:51.000000000 +0200
+++ pesign-113/src/pesign.sysvinit.in	2019-09-12 19:03:31.078192655 +0200
@@ -13,45 +13,47 @@
 # Description: The pesign PE signing daemon
 ### END INIT INFO
 
-. /etc/init.d/functions
-[ -f /usr/bin/pesign ] || exit 1
+. /etc/rc.d/init.d/functions
 
 RETVAL=0
 
-start(){
-    echo -n "Starting pesign: "
+start() {
+    if [ -f /var/lock/subsys/pesign ]; then
+	msg_already_running "pesign"
+	return
+    fi
+
+    msg_starting pesign
     mkdir /var/run/pesign 2>/dev/null &&
         chown pesign:pesign /var/run/pesign &&
         chmod 0770 /var/run/pesign
     daemon /usr/bin/pesign --daemonize
     RETVAL=$?
-    echo
-    touch /var/lock/subsys/pesign
+    [ $RETVAL -eq 0 ] && touch /var/lock/subsys/pesign
     @@LIBEXECDIR@@/pesign/pesign-authorize
 }
 
-stop(){
-    echo -n "Stopping pesign: "
+stop() {
+    if [ ! -f /var/lock/subsys/pesign ]; then
+	msg_not_running pesign
+	return
+    fi
+
+    msg_stopping pesign
     killproc -p /var/run/pesign.pid pesignd
-    RETVAL=$?
-    echo
     rm -f /var/lock/subsys/pesign
 }
 
-restart(){
-    stop
-    start
-}
-
-reload(){
+condrestart() {
+    if [ ! -f /var/lock/subsys/pesign ]; then
+	msg_not_running pesign
+	RETVAL=$1
+	return
+    fi
     stop
     start
 }
 
-condrestart(){
-    [ -e /var/lock/subsys/pesign ] && restart
-}
-
 # See how we were called.
 case "$1" in
     start)
@@ -61,26 +63,22 @@
 	stop
 	;;
     status)
-	status /usr/bin/pesign
+	status pesign /usr/bin/pesign
+	RETVAL=$?
 	;;
     restart)
-	restart
-	;;
-    reload)
-	reload
+	stop
+	start
 	;;
     force-reload)
-	reload
-	;;
-    condrestart)
-	condrestart
+	condrestart 7
 	;;
-    try-restart)
-	condrestart
+    try-restart|condrestart)
+	condrestart 0
 	;;
     *)
-	echo "Usage: pesign {start|stop|status|restart|condrestart|reload}"
-	RETVAL=1
+	echo "Usage: pesign {start|stop|status|restart|try-restart|condrestart|force-reload}"
+	exit 3
 esac
 
 exit $RETVAL
